rnn/island.cxx

```diff
@@ -348,14 +348,12 @@ int32_t Island::get_status() {
 }
 
 void Island::set_status(int32_t status_to_set) {
-    if(status_to_set >= 0 || status_to_set <= 2){
+    if(status_to_set >= 0 && status_to_set <= 2) {
```

Comment:
@travisdesell this was meant to be inclusive right? the `||` lets any value through.

Reply:
Yeah that should be inclusive. Also, there should be a space after the if to keep the formatting nice.  I think we should actually not have hardcoded 0 and 2 here, but rather check for existing defined constant values.

Reply:
Done. Do you prefer `#define N_TYPES (0)`... over `enum`?

Reply:
actually yeah that might be better! i guess my worry is if the number of status's change that this will break.

Reply:
I actually just put in each of the status types, as that way if a new one gets added and this isn't updated then it will crash with an error.